docs/adr/environments-and-ci.md

```diff
@@ -0,0 +1,78 @@
+# ADR 35580f0d-f429-412b-acef-83655e3cab11: Runtime/engine/host/environment support and CI
+
+## Status
+
+Proposed
+
+## Submitters
+
+- @ctcpip
+
+## Decision Owners
+
+- @expressjs/express-tc
+
+## Context
+
+Express and its libraries were specifically designed to run with Node.js (V8). While some of our libraries can run in other environments (e.g. runtimes, engines, browsers), they are not necessarily supported in all environments. Consequently, our CI systems do not include other environments as part of their testing workflows.
+
+Several points were raised during the discussion:
+
+- Cost of running additional CI vs the likelihood of detecting a problem
+- Introducing maintenance overhead and possible coupling to other environments' development lifecycle
+- No JS engine implements ECMAScript 100% correctly; thus, claiming "ES2015 support" does not guarantee correctness across all environments.
+- Environment regressions or language edge cases could break functionality in unpredictable ways that are not practical for us to monitor across all environments.
+
+## Decision
+
+- We will **not** add non-Node.js environment testing to our CI pipelines.
+  - Exceptions may be made iff there is strongly compelling, project-aligned justification.
+- CI will continue to run only against supported Node.js versions.
+- Support for other environments may exist, per maintainer discretion, but we do not guarantee support across all environments.
+  - Some libraries, particularly language-only libraries which do not require non-language APIs, strive to support as many environments as possible.
+    - Nonetheless, support is not guaranteed across every possible environment, and is provided on a best-effort basis.
```

Comment:
```suggestion
- Some Express libraries may work in other environments (e.g., Bun, Deno, Cloudflare Workers), but we do not guarantee compatibility or prioritize testing and development for them. Support outside Node.js is best-effort and may vary between packages, depending on maintainer interest and alignment with project goals.
  - Some libraries, particularly language-only libraries which do not require non-language APIs, strive to support as many environments as possible.
```

Reply:
intent is to deliberately avoid mentioning specific environments

Reply:
https://github.com/expressjs/discussions/pull/365#discussion_r2067487965

Reply:
edited out the mention